tls/s2n_client_hello.c

```diff
@@ -348,17 +351,20 @@ static S2N_RESULT s2n_client_hello_process_cb_response(struct s2n_connection *co
 
 int s2n_client_hello_recv(struct s2n_connection *conn)
 {
-    POSIX_ENSURE(conn->client_hello.callback_async_blocked == 0, S2N_ERR_ASYNC_BLOCKED);
+    if (conn->client_hello.callback_enable_poll == 0) {
+        POSIX_ENSURE(conn->client_hello.callback_async_blocked == 0, S2N_ERR_ASYNC_BLOCKED);
+    }
```

Comment:
So you're really trying to say:
```
    POSIX_ENSURE(!conn->client_hello.callback_async_blocked || !conn->client_hello.callback_enable_poll, S2N_ERR_ASYNC_BLOCKED);
```
It might be cleaner to define a little bool helper method / macro to name that combo of checks. Like s2n_client_hello_is_async_blocked.

Reply:
currently we get ERR only when:
```
poll async     ERR
0    0         1 
0    1         0
1    1         0
1    0         0
```

with `POSIX_ENSURE(!conn->client_hello.callback_async_blocked || !conn->client_hello.callback_enable_poll, S2N_ERR_ASYNC_BLOCKED);`

we get:
```
poll async     ERR
0    0         1
0    1         1
1    1         0
1    0         1
```